### PR TITLE
Update CHANGELOG to include all versions

### PR DESCRIPTION
**Pull request by Hoverbear**
###  What have you changed?

Update our CHANGELOG to include all versions, in semver order.

**Note for reviewers:** The copy of the changelog is already established and it is not up for review (it's a different PR), this PR is only related to filling in missing gaps and adding dates/styling consistency.

###  What is the type of the changes?

- Engineering (engineering change which doesn't change any feature or fix any issue)

###  How is the PR tested?

Visual

###  Does this PR affect documentation (docs) or should it be mentioned in the release notes?

Yes this updates the CHANGELOG.

###  Does this PR affect `tidb-ansible`?

Nope

### CODE REVIEW

**Comment by siddontang** [comment]
PTAL @dcalvin

File: CHANGELOG.md [hunk 1]
```diff
@@ -5,98 +5,112 @@ See also [TiDB Changelog](https://github.com/pingcap/tidb/blob/master/CHANGELOG.
...
-
 + Tikv-ctl 
     - Improve `bad-regions` and `tombstone` subcommands. [4862](https://github.com/tikv/tikv/pull/4862)
-
 + Misc
     - Add dist_release. [4841](https://github.com/tikv/tikv/pull/4841)
 
-## [3.0.0-rc.2]
-
+## [3.0.0-rc.2] -2019-05-28
```
Comment (dcalvin) [inline CHANGELOG.md]:
```suggestion
## [3.0.0-rc.2] - 2019-05-28
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -5,98 +5,112 @@ See also [TiDB Changelog](https://github.com/pingcap/tidb/blob/master/CHANGELOG.
...
++ Fix potential scheduling jitters caused by premature `PutStore` [#5277](https://github.com/tikv/tikv/pull/5277)
++ Fix incorrect timestamps reported from Region heartbeats [#5296](https://github.com/tikv/tikv/pull/5296)
++ Fix potential TiKV panics during region merge [#5291](https://github.com/tikv/tikv/pull/5291)
++ Speed up leader change check for the dead lock detector [#5317](https://github.com/tikv/tikv/pull/5317)
++ Support using `grpc env` to create deadlock clients [#5346](https://github.com/tikv/tikv/pull/5346)
++ Add `config-check` to check whether the configuration is correct [#5349](https://github.com/tikv/tikv/pull/5349)
++ Fix the issue that ReadIndex does not return anything when there is no leader [#5351](https://github.com/tikv/tikv/pull/5351)
++ Exclude shared block cache from core dump [#5322](https://github.com/tikv/tikv/pull/5322)
+
+## [3.0.2] -2019-08-06
```
Comment (dcalvin) [inline CHANGELOG.md]:
```suggestion
## [3.0.2] - 2019-08-06
```

**Comment by Hoverbear** [comment]
@dcalvin Thanks!

**Pull request merged**